x/nft/keeper/keeper.go

```diff
@@ -159,3 +159,22 @@ func (k Keeper) BurnNFT(ctx sdk.Context, denomID, tokenID string, owner sdk.AccA
 
 	return nil
 }
+
+// BurnNFTUnverified deletes a specified NFT without verifying if the owner is the creator of denom
+// Needed for IBC transfer of NFT
+func (k Keeper) BurnNFTUnverified(ctx sdk.Context, denomID, tokenID string, owner sdk.AccAddress) error {
```

Comment:
may I ask if we also need check `HasNFT` here?

Reply:
`IsOwner` will check for this.